core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterPersistenceEngine.scala

```diff
@@ -53,9 +53,12 @@ private[spark] trait MesosClusterPersistenceEngine {
  * all of them reuses the same connection pool.
  */
 private[spark] class ZookeeperMesosClusterPersistenceEngineFactory(conf: SparkConf)
-  extends MesosClusterPersistenceEngineFactory(conf) {
+  extends MesosClusterPersistenceEngineFactory(conf) with Logging {
 
-  lazy val zk = SparkCuratorUtil.newClient(conf, "spark.mesos.deploy.zookeeper.url")
+  // TODO(tnachen): Remove support for spark.mesos.deploy.zookeeper.url in 0.28.
```

Comment:
same here, no need to do this later. Just do it now. (also this is Spark so you should use Spark versions in comments, not Mesos versions)